CHANGELOG.md

```diff
@@ -167,12 +167,16 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 
 - Deprecated the `TestTubeLogger` ([#9065](https://github.com/PyTorchLightning/pytorch-lightning/pull/9065))
 
+
 - Deprecated `on_{train/val/test/predict}_dataloader()` from `LightningModule` and `LightningDataModule` [#9098](https://github.com/PyTorchLightning/pytorch-lightning/pull/9098)
 
 
 - Updated deprecation of  `argparse_utils.py` from removal in 1.4 to 2.0 ([#9162](https://github.com/PyTorchLightning/pytorch-lightning/pull/9162))
 
 
+- Deprecated `on_keyboard_interrupt` callback hook ([#9183](https://github.com/PyTorchLightning/pytorch-lightning/pull/9183))
```

Comment:
```suggestion
- Deprecated `on_keyboard_interrupt` callback hook in favor of new `on_exception` hook ([#9183](https://github.com/PyTorchLightning/pytorch-lightning/pull/9183))
```

Reply:
@awaelchli should we add this to the contributing section?  Something like "what to do if you're deprecating something?" 

Includes how to even initiate the process, what type of sign-off or approval is needed, and then the mechanics of emitting warnings, adding tests, and making sure the change log reflects what to use instead? 

Then the governance docs around API lifecycle could remain the high-level strategy around this

Reply:
resolving to allow for merge